src/pass.h

```diff
@@ -297,6 +291,14 @@ class Pass {
   // out any Stack IR - it would need to be regenerated and optimized.
   virtual bool modifiesBinaryenIR() { return true; }
 
+  // Whether this pass operates on StackIR. Once StackIR is generated, the
+  // normal Binaryen IR is no longer valid, so it is an error to run a
+  // non-StackIR pass after that point.
+  virtual bool acceptsStackIR() { return false; }
+
+  // Whether this pass operates on nested IR.
+  virtual bool acceptsNestedIR() { return true; }
```

Comment:
I'm not sure. I think for this PR it probably makes more sense to stick to the `StackIR`/`BinaryenIR` naming. In the future if Binaryen IR ends up getting a flat form to replace Stack IR we can figure out a better naming scheme.

Reply:
I think StackIR/BinaryenIR are the terms we've used so far. Makes sense to me to continue with those.

Reply:
What I meant was, then why do have two different names (BinaryenIR and NestedIR) for the same thing?

Reply:
The PR originally had two different names because I couldn't decide whether I wanted to rename it or not so I only renamed it in half the relevant places. My bad 😅